code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -295,22 +303,27 @@ paths:
     delete:
       tags:
         - Application
-      summary: Delete an Application from an OP
-      description: Delete all the information and content related to an Application
-      operationId: deleteApp
+      summary: Terminate an Application Instance
+      description: Terminate a running instance of an application within an Edge Cloud Zone
+      operationId: deleteAppInstance
       parameters:
         - name: appId
+          description: A globally unique identifier associated with the application. OP generates this identifier when the application is submitted over NBI.
           in: path
-          description: Identificator of the application to be deleted provided by the OP NBI once the submission was successful
           required: true
+          schema: 
+              $ref: '#/components/schemas/AppId'
+        - name: appInstanceId
+          in: query
+          description: Identificator of the specific application instance that will be terminated
+          required: false
```

Comment:
AppId is a common property for all the AppInstances, so DELETE method will apply for all the instances with the same AppId and therefore you don't have a 404. But is true that it would bring some complexity in terms of knowing which instance has been deleted and which no, but to check that, you can use the GET method. In any case if we think that it makes more sense to DELETE just one instance per call, we can change it to `DELETE /app/{appId}/instance/appinstance/{appInstanceId}.` @gunjald, @nicolacdnll, @Kevsy @FabrizioMoggio  WDYT?

Reply:
I believed that this was a synchronous call to label all instances's `AppInstanceInfo.status=terminating`, and instances can take some time to be deleted. If this is the case, I'm fine to have one call to flag all instances to be terminated. I deduct this from the 202 response.

If instead, I misunderstood, and the call is to synchronously terminate all instances, I agree with Jon.

In any case, I believe the description should be improved to clearly describe the expected behavior and prevent confusion.

Reply:
I think that typically removing all instances in a single call may lead to indeterministic behaviors at times and it is difficult to aggregate the results of termination status of each instances as a single status. I would also recommend to use DELETE /app/{appId}/instance/appinstance/{appInstanceId} with appInstanceId as mandatory path parameter.

Reply:
Agree with you guys, for this MVP only DELETE /app/{appId}/instance/appinstance/{appInstanceId} will be allow. 
But in next releases we could discuss how to enable a full deletion.